### ISSUE TITLE
List - when using keyboard, active list item isn't always in view

### ISSUE THREAD

**Issue by taysea** [grommet/grommet#7323]
### Expected Behavior

When using arrow keys to get to list item, would expect active list item to be in view on the screen.

### Actual Behavior

Active list item can end up being off screen but page doesn't scroll down.

### URL, screen shot, or Codepen exhibiting the issue

https://storybook.grommet.io/?path=/story/visualizations-list-onclickitem--on-click-item-list

https://github.com/user-attachments/assets/b5ca4265-db00-4132-8cda-89b263866b01

### Steps to Reproduce

1. Make window smaller than length of list
2. Tab to List
3. Use arrow keys to move down

### Your Environment

- Grommet version: Latest (on stable of storybook)
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): MacOS

**Comment by vijaykr338** [comment]
I'd like to be assigned on this

**Comment by jcfilben** [comment]
Hi @vijaykr338 feel free to work on this

**Comment by vijaykr338** [comment]
I fixed the issue, using the scrollIntoView method, by attaching a ref to each currently active element

in src/js/components/List/List.js

`const activeItemRef = useRef(null);
    
    const setActiveItemRef = (element) => {
      if(element) {
        element.scrollIntoView({
          behaviour: 'smooth',
          block: 'nearest',
        });
      }
      activeItemRef.current = element;
    }`

then simply attaching a ref to each item in the list 
`  return (
                  <StyledItem
                    key={key}
                    tag="li"
                    background={adjustedBackground}
                    border={adjustedBorder}
                    isDisabled={isDisabled}
                    ref={index === active ? setActiveItemRef : undefined}
                    flex={false}
                    pad={pad || theme.list.item.pad}
                    {...defaultItemProps}
                    {...boxProps}
                    {...clickProps}
                    {...orderProps}
                    {...itemAriaProps}
                    {...passThemeFlag}
                  >
                    {showIndex && onOrder && (
                      <Text color={pinnedColor}>{index + 1}</Text>
                    )}
                    {content}
                    {displayPinned}
                    {orderControls}
                  </StyledItem>
                );`
                
                
                but it is causing way too many errors to fail when I run yarn test

**Comment by vijaykr338** [comment]
I fixed the issue, using the scrollIntoView method, by attaching a ref to each currently active element
but it is causing many errors to occur

`  const activeItemRef = useRef(null); `

```  const setActiveItemRef = (element) => {
      if (element) {
        element.scrollIntoView({
          behaviour: 'smooth',
          block: 'nearest',
        });
      }
      activeItemRef.current = element;
    };
```
now when I am returning I am simply attaching the ref
``` <StyledItem
                    key={key}
                    tag="li"
                    background={adjustedBackground}
                    border={adjustedBorder}
                    isDisabled={isDisabled}
                    ref={index === active ? setActiveItemRef : undefined}
                    flex={false}
                    pad={pad || theme.list.item.pad}
                    {...defaultItemProps}
                    {...boxProps}
                    {...clickProps}
                    {...orderProps}
                    {...itemAriaProps}
                    {...passThemeFlag}
                  >
                    {showIndex && onOrder && (
                      <Text color={pinnedColor}>{index + 1}</Text>
                    )}
                    {content}
                    {displayPinned}
                    {orderControls}
                  </StyledItem>
                  
                  ```
